gcp/rakefiles/secrets.rb

```diff
@@ -0,0 +1,39 @@
+require "securerandom"
+require "yaml"
+
+class Secrets
+
+  def self.set_secrets()
+    saved_secrets_file_path = "../#{ENV['ENV']}/secrets/#{ENV["TF_VAR_project_id"]}-secrets.yml"
+
+    begin
+      @secrets = YAML.load(File.read(saved_secrets_file_path))
```

Comment:
Now that this is a proper ruby function that returns the hash it constructs, I think these should all be `secrets` not `@secrets`. I think it's best to assign to the instance variable `@secrets` on the calling end, as you are already doing in entrypoint.rake.